File: backend/internal/db/commands/recipeCommands.go
package commands

import (
	"github.com/georgysavva/scany/pgxscan"
	"github.com/google/uuid"
	"github.com/jackc/pgx/v4"
	"github.com/viddem/vrecipes/backend/internal/db/tables"
)

var createRecipeCommand = `
INSERT INTO recipe(name, unique_name, description, oven_temp, estimated_time, deleted, created_by, portions, portions_suffix)
		   VALUES ($1,   $2,          $3,          $4,        $5,             $6,	   $7,         $8,       $9)
RETURNING id, name, unique_name, description, oven_temp, estimated_time, deleted, created_by, portions, portions_suffix`

func CreateRecipe(tx pgx.Tx, name, uniqueName, description, portionsSuffix string, ovenTemp, estimatedTime, portions int, createdBy uuid.UUID) (*tables.Recipe, error) {
	var recipe tables.Recipe
	err := pgxscan.Get(ctx, tx, &recipe, createRecipeCommand, name, uniqueName, description, ovenTemp, estimatedTime, false, createdBy, portions, portionsSuffix)
	return &recipe, err
}

var updateRecipeCommand = `
UPDATE recipe 
SET name=$1,
	unique_name=$2,
	description=$3,
	oven_temp=$4,
	estimated_time=$5,
	portions=$6,
	portions_suffix=$7
WHERE id=$8
`

func UpdateRecipe(tx pgx.Tx, name, uniqueName, description string, ovenTemp, estimatedTime, portions int, portionsSuffix string, recipeId uuid.UUID) error {
	_, err := tx.Exec(ctx, updateRecipeCommand, name, uniqueName, description,
		ovenTemp, estimatedTime, portions, portionsSuffix, recipeId)
	return err
}

var recipeSetDeletedCommand = `
UPDATE recipe
SET deleted=true,
	name=$1,
	unique_name=$2
WHERE id=$3
`

func RecipeSetDeleted(tx pgx.Tx, name, uniqueName string, id uuid.UUID) error {
	_, err := tx.Exec(ctx, recipeSetDeletedCommand, name, uniqueName, id)
	return err
}

File: python_backend/src/arch/command/RecipeCommands.py
from uuid import UUID

from pony.orm import db_session

from db import Recipe


@db_session
def insert_new_recipe(name: str, unique_name: str, description: str = "", oven_temp: int = -1,
                      estimated_time: int = -1) -> Recipe:
    """
    Create a new recipe.
    :return: the new recipe
    """
    return Recipe(name=name, unique_name=unique_name, description=description, oven_temp=oven_temp,
                  estimated_time=estimated_time)


@db_session
def update_recipe_name(id: UUID, name: str, unique_name: str):
    recipe = Recipe.get(id=id)
    recipe.name = name
    recipe.unique_name = unique_name


@db_session
def update_recipe_general(id: UUID, description: str, oven_temp: int, estimed_time: int):
    recipe = Recipe.get(id=id)
    recipe.description = description
    recipe.oven_temp = oven_temp
    recipe.estimated_time = estimed_time
